packages/schematics/angular/application/index.ts

```diff
@@ -300,8 +300,8 @@ export default function (options: ApplicationOptions): Rule {
     const isRootApp = options.projectRoot !== undefined;
     const appDir = isRootApp
       ? normalize(options.projectRoot || '')
-      : join(normalize(newProjectRoot), options.name);
-    const sourceDir = `${appDir}/src/app`;
+      : join(normalize(newProjectRoot), strings.dasherize(options.name));
+    const sourceDir = `${strings.dasherize(appDir)}/src/app`;
```

Comment:
```suggestion
    const sourceDir = `$(appDir}/src/app`;
```

Shouldn't be needed, since it's already transformed in the above line.